bot/exts/code_jams/_flows.py

```diff
@@ -213,3 +213,58 @@ async def remove_flow(bot: SirRobin, member: discord.Member, ctx: commands.Conte
         if role.name == TEAM_LEADER_ROLE_NAME:
             await member.remove_roles(role)
     await ctx.send(f"Successfully removed {member.mention} from team {team['team']['name']}")
+
+
+async def pin_flow(
+        ctx: commands.Context,
+        roles: tuple[int, ...],
+        mgmt_api: APIClient,
+        message: Optional[discord.Message] = None,
+        unpin: bool = False
+) -> None:
+    """
+    Pin or unpin the given message.
+
+    Additional checks have been put in place, to ensure
+    messages can only be (un)pinned inside the Code Jam Category
+    by the Events Team and Admins, and participants can
+    only (un)pin messages in their own team channel.
+    """
+    referenced_message = getattr(ctx.message.reference, "resolved", None) or message
+    pin_msg = f"{'un' if unpin else ''}pin"
+    if not isinstance(referenced_message, discord.Message):
+        await ctx.reply(
+            ":x: You have to either reply to a message or provide a message link / message id"
+            f" in order to {pin_msg} it."
+        )
+        return
+    if referenced_message.channel != ctx.channel:
+        await ctx.reply(f":x: You cannot {pin_msg} a message outside of this team's channel.")
+        return
+
+    if referenced_message.pinned and not unpin:
+        await ctx.reply(":x: The message has already been pinned!")
+        return
+    elif not referenced_message.pinned and unpin:
+        await ctx.reply(":x: The message has already been unpinned!")
+        return
+
+    if any(role.id in roles for role in getattr(ctx.author, "roles", [])):
+        await _creation_utils.pin_message(referenced_message, ctx, unpin)
+        return
+    try:
+        team = await mgmt_api.get(
+            f"users/{ctx.author.id}/current_team",
+            raise_for_status=True
+        )
+    except ResponseCodeError as err:
+        if err.response.status == 404:
+            await ctx.reply(":x: It seems like you're not a participant!")
+        else:
+            await ctx.reply("Something went wrong while processing the request! We have notified the team!")
+            log.error(f"Something went wrong with processing the request! {err}")
+    else:
+        if ctx.channel.id == int(team["team"]["discord_channel_id"]):
+            await _creation_utils.pin_message(referenced_message, ctx, unpin=unpin)
+        else:
+            await ctx.reply(f"You don't have permission to {pin_msg} this message in that channel!")
```

Comment:
```suggestion
            await ctx.reply(f"You don't have permission to {pin_msg} this message in this channel!")
```
Since the user can only use this command on a message in their current channel "this" makes more sense.

Reply:
Patched.